idea/src/org/jetbrains/kotlin/idea/quickfix/RemoveUselessCastFix.kt

```diff
@@ -26,7 +26,7 @@ class RemoveUselessCastFix(element: KtBinaryExpressionWithTypeRHS) : KotlinQuick
     }
 
     companion object : KotlinSingleIntentionActionFactory() {
-        operator fun invoke(element: KtBinaryExpressionWithTypeRHS) = dropEnclosingParenthesesIfPossible(element.replaced(element.left))
+        operator fun invoke(element: KtBinaryExpressionWithTypeRHS) = element.replaced(element.left).dropEnclosingParenthesesIfPossible()
```

Comment:
We can use `KtPsiUtil.deparenthesize` as it removes all parentheses, not only single one

Reply:
Note that `KtPsiUtil.deparenthesize()` accepts the parent `KtParenthesizedExpression`, not the inner expression. `element` here is the inner expression.

The purpose of `dropEnclosingParenthesesIfPossible()` here is a case like this:
```kt
val x = (1 ?: null)
```
When we remove the `?: null` and we are left with `(1)`, the surrounding parentheses is useless and we should remain with `val x = 1`. This is why there is a call to `KtPsiUtil.areParenthesesUseless` in `dropEnclosingParenthesesIfPossible`.

In some cases, the parentheses are not useless, so we can't just remove all of them:
```kt
fun foo() {}
foo()
({} ?: null)
```
If we replaced the last line with `{}` instead of `({})` then it will be treated as a trailing lambda argument to `foo()`.

What I _can_ do is keep dropping unnecessary parentheses if they still exist, so that multiple nested parentheses will be dropped in `val x = (((1 ?: null)))`. I will add this modification.